static/app/utils/useResizable.tsx

```diff
@@ -0,0 +1,170 @@
+ *
+ * Currently only supports resizing width and not height.
+ */
+export const useResizable = ({
+  ref,
+  initialSize = RESIZABLE_DEFAULT_WIDTH,
+  maxWidth = RESIZABLE_MAX_WIDTH,
+  minWidth = RESIZABLE_MIN_WIDTH,
+  onResizeEnd,
+  onResizeStart,
+  sizeStorageKey,
+}: UseResizableOptions): {
+  /**
+   * Whether the drag handle is held.
+   */
+  isHeld: boolean;
+  /**
+   * Apply this to the drag handle element to include 'reset' functionality.
+   */
+  onDoubleClick: () => void;
+  /**
+   * Attach this to the drag handle element's onMouseDown handler.
+   */
+  onMouseDown: (e: React.MouseEvent) => void;
+  /**
+   * The current size of the container. This is NOT updated during the drag
+   * event, only after the user finishes dragging.
+   */
+  size: number;
+} => {
+  const [isHeld, setIsHeld] = useState(false);
+
+  const isDraggingRef = useRef<boolean>(false);
+  const startXRef = useRef<number>(0);
+  const startWidthRef = useRef<number>(0);
+
+  useEffect(() => {
+    if (ref.current) {
+      const storedSize = sizeStorageKey
+        ? parseInt(localStorage.getItem(sizeStorageKey) ?? '', 10)
+        : undefined;
+
+      ref.current.style.width = `${storedSize ?? initialSize}px`;
+    }
+  }, [ref, initialSize, sizeStorageKey]);
+
+  const handleMouseDown = useCallback(
+    (e: React.MouseEvent) => {
+      setIsHeld(true);
+      e.preventDefault();
+
+      const currentWidth = ref.current
+        ? parseInt(getComputedStyle(ref.current).width, 10)
+        : 0;
+
+      isDraggingRef.current = true;
+      startXRef.current = e.clientX;
+      startWidthRef.current = currentWidth;
+
+      document.body.style.cursor = 'ew-resize';
+      document.body.style.userSelect = 'none';
+      onResizeStart?.();
+    },
+    [ref, onResizeStart]
+  );
+
+  const handleMouseMove = useCallback(
+    (e: MouseEvent) => {
+      if (!isDraggingRef.current) return;
+
+      const deltaX = e.clientX - startXRef.current;
+      const newWidth = Math.max(
+        minWidth,
+        Math.min(maxWidth, startWidthRef.current + deltaX)
+      );
+
+      if (ref.current) {
+        ref.current.style.width = `${newWidth}px`;
+      }
+    },
+    [ref, minWidth, maxWidth]
+  );
+
+  const handleMouseUp = useCallback(() => {
+    setIsHeld(false);
+    const newSize = ref.current?.offsetWidth ?? initialSize;
+    isDraggingRef.current = false;
+    document.body.style.cursor = '';
+    document.body.style.userSelect = '';
+    onResizeEnd?.(newSize);
+    if (sizeStorageKey) {
+      localStorage.setItem(sizeStorageKey, newSize.toString());
+    }
+  }, [onResizeEnd, ref, sizeStorageKey, initialSize]);
+
+  useEffect(() => {
+    document.addEventListener('mousemove', handleMouseMove);
+    document.addEventListener('mouseup', handleMouseUp);
+
+    return () => {
+      document.removeEventListener('mousemove', handleMouseMove);
+      document.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, [handleMouseMove, handleMouseUp]);
```

Comment:
I don't think we want these listeners on the document all the time. These should only be attached once we're holding. Should be as simple as wrapping this in `if (isHeld)`